spec/model/post_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'rails_helper'
+RSpec.describe Post, type: :model do
+  subject { Post.new(title: 'My First Post', comments_counter: 4, likes_counter: 20) }
+
+  before { subject.save }
+
+  it 'title should be present' do
+    subject.title = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'comments_counter should be present' do
+    subject.comments_counter = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'likes_counter should be present' do
+    subject.likes_counter = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'title should not be too long' do
+    subject.title = 'a' * 100
+    expect(subject).to_not be_valid
+  end
+
+  it 'comments_counter should be integer' do
+    subject.comments_counter = 'b'
+    expect(subject).to_not be_valid
+  end
+
+  it 'comments_counter should be greater than or equal to zero' do
+    subject.comments_counter = -1
+    expect(subject).to_not be_valid
+  end
+
+  it 'likes_counter should be integer' do
+    subject.likes_counter = 'i'
+    expect(subject).to_not be_valid
+  end
+
+  it 'likes_counter should be greater than or equal to zero' do
+    subject.likes_counter = -1
+    expect(subject).to_not be_valid
+  end
+end
```

Comment:
- [ ] All the tests contained here are testing for negative scenarios. Kindly add one or more scenarios for when the Post is valid to account for all possible scenarios.

- [ ] Kindly add tests for the `Comment` and `Like` models.